File: lib/feature/ddayFeature/domain/entities/dday_entity.dart
import 'package:realm/realm.dart';

part 'dday_entity.realm.dart';

@RealmModel()
class _DdayEntity {
  @PrimaryKey()
  late final ObjectId id;

  late String title;
  late DateTime date;
  //-----schema 2
  late bool repeatAnniversary;
  late int notificationType;
  //schema 2-----
  late bool dayPlus;
}

File: android/app/src/main/kotlin/io/github/yyyng2/make_my_day/DdayEntity.kt
package io.github.yyyng2.make_my_day

import io.realm.kotlin.types.RealmObject
import io.realm.kotlin.types.annotations.PrimaryKey
import io.realm.kotlin.types.RealmInstant
import org.mongodb.kbson.ObjectId
import java.util.Date

class DdayEntity : RealmObject {
    @PrimaryKey
    var id: ObjectId = ObjectId()
    var title: String = ""
    var date: RealmInstant = RealmInstant.from(0, 0)
    var dayPlus: Boolean = false
    var repeatAnniversary: Boolean = false
    var notificationType: Int = 0
}